extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -198,14 +198,14 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
-   * Returns true if this element is the child of an amp-ad element.  For use by
+   * Returns true if this element was loaded from an amp-ad element.  For use by
    * network-specific implementations that don't want to allow themselves to be
    * embedded directly into a page.
    * @return {boolean}
    */
   isInAmpAdTag() {
-    return !!this.element.parentElement &&
-        this.element.parentElement.tagName == 'AMP-AD';
+    return this.element.tagName == 'AMP-AD' ||
```

Comment:
Not necessarily.  Other ad networks may allow for placing the impl extension (e.g. amp-ad-network-${type}-impl element) directly on the publisher page.  We want to make sure that this is not supported for adsense/doubleclick thus this check.